support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20RevocationEndpointController.java

```diff
@@ -0,0 +1,191 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.view.json.MappingJackson2JsonView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This is {@link OAuth20RevocationEndpointController}.
+ *
+ * @author Julien Huon
+ * @since 6.2.0
+ */
+@Slf4j
+public class OAuth20RevocationEndpointController extends BaseOAuth20Controller {
+    public OAuth20RevocationEndpointController(final OAuth20ConfigurationContext oAuthConfigurationContext) {
+        super(oAuthConfigurationContext);
+    }
+
+    /**
+     * Handle request for revocation.
+     *
+     * @param request  the request
+     * @param response the response
+     * @return the response entity
+     */
+    @PostMapping(path = '/' + OAuth20Constants.BASE_OAUTH20_URL + '/' + OAuth20Constants.REVOCATION_URL,
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    public ModelAndView handleRequest(final HttpServletRequest request,
+                                      final HttpServletResponse response) {
+        val context = new JEEContext(request, response, getOAuthConfigurationContext().getSessionStore());
+
+        if (!verifyRevocationRequest(context)) {
+            LOGGER.error("Revocation request verification failed. Request is missing required parameters");
+            return OAuth20Utils.writeError(response, OAuth20Constants.INVALID_REQUEST);
+        }
+
+        val manager = new ProfileManager<CommonProfile>(context, context.getSessionStore());
+        val clientId = OAuth20Utils.getClientIdAndClientSecret(context).getLeft();
+        val registeredService = getRegisteredServiceByClientId(clientId);
+
+        if (isServiceNeedAuthentication(registeredService)) {
+            if (manager.get(true).isEmpty()) {
+                LOGGER.error("Service [{}] requests authentication", clientId);
+                return OAuth20Utils.writeError(response, OAuth20Constants.ACCESS_DENIED);
+            }
+        }
+        val token = context.getRequestParameter(OAuth20Constants.TOKEN)
+            .map(String::valueOf).orElse(StringUtils.EMPTY);
+
+        if (isAsupportedTokenType(token)) {
+            return generateRevocationResponse(token, clientId, response);
+        }
+        return OAuth20Utils.writeError(response, OAuth20Constants.INVALID_REQUEST);
+    }
+
+    /**
+     * Generate revocation token response.
+     *
+     * @param token the token to revoke
+     * @param client the client who requests the revocation
+     * @param response the response
+     * @return the model and view
+     */
+    private ModelAndView generateRevocationResponse(final String token,
+                                                    final String client,
+                                                    final HttpServletResponse response) {
+
+        if (token.startsWith(OAuth20RefreshToken.PREFIX)) {
```

Comment:
This sort of thing can never pass. Comparing ticket ids by prefixes is generally a malpractice and should be avoided.

Reply:
Ok, sorry about that.

Reply:
Done.